template.tpl

```diff
@@ -246,11 +242,11 @@ const generateEventName = (requestData) => {
     requestData.event_name = 'click';
   } else if (requestData.download) {
     requestData.event_name = 'file_download';
-  } else if (requestData.action_name) {
-    requestData.event_name = 'page_view';
   } else if (requestData.idgoal) {
```

Comment:
@sahava was there a reason why we added goal conversions so low in the event detection pipeline?

> The first thing the tracker does to detect the type of event it’s dealing with is look for signs that the event resulted in a conversion. If the tracked event has an &idgoal parameter with a value other than 0, it is classified as Goal conversion

https://help.piwik.pro/support/questions/what-are-events-and-how-are-they-detected/#step-1:-goal-conversion

Looking at the docs - I'd move it to the top and change the condition:

```javascript
const generateEventName = (requestData) => {
  if (requestData.event_name) {
    return requestData;
  } else if (requestData.idgoal && requestData.idgoal !== 0) {
    requestData.event_name = 'goal_conversion';
  } else if (requestData.ping) {
    requestData.event_name = 'ping';
  } else if (requestData.c_i) {
    requestData.event_name = 'content_interaction';
  } else if (requestData.c_n) {
    requestData.event_name = 'content_impression';
  } else if (requestData.e_t) {
    requestData.event_name = requestData.e_t;
  } else if (requestData.e_c) {
    requestData.event_name = requestData.e_c;
  } else if (requestData.search) {
    requestData.event_name = 'search';
  } else if (requestData.link) {
    requestData.event_name = 'click';
  } else if (requestData.download) {
    requestData.event_name = 'file_download';
  } else if (requestData.action_name) {
    requestData.event_name = 'page_view';
  }  else {
    requestData.event_name = DEFAULT_EVENT_NAME;
  }
  return requestData;
};
```